packages/block-editor/src/components/plain-text/content.scss

```diff
@@ -1,11 +1,27 @@
-.block-editor-plain-text {
+.block-library-html__edit .block-editor-plain-text.components-base-control {
+	padding: 0 !important;
+}
+
+.block-editor-plain-text textarea {
```

Comment:
This is necessary because `TextareaControl` has a couple of wrappers around the `textarea` element.